# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/mozjs/hotfix-ia64-check-textrel.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

# ia64 somehow has text relocations in libmozjs*.so (FIXME), allow it also in
# check_binary.py (-z notext is already stripped by wrapper)

--- firefox-140.0/python/mozbuild/mozbuild/action/check_binary.py.vanilla	2026-07-13 22:00:00.000000000 +0200
+++ firefox-140.0/python/mozbuild/mozbuild/action/check_binary.py	2026-07-13 22:05:00.000000000 +0200
@@ -12,6 +12,10 @@
 from mozpack.executables import ELF, UNKNOWN, get_type
 from packaging.version import Version
 
+# ia64's GP-relative addressing range is too small for libxul.so; the
+# linker produces text relocations regardless of -z text/notext.
+IS_IA64 = buildconfig.substs.get("TARGET_CPU") == "ia64"
+
 from mozbuild.util import memoize
 
 STDCXX_MAX_VERSION = Version("3.4.19")
@@ -286,7 +290,8 @@
 
     # Disabled for local builds because of readelf performance: See bug 1472496
     if not buildconfig.substs.get("DEVELOPER_OPTIONS"):
-        checks.append(check_textrel)
+        if not IS_IA64:
+            checks.append(check_textrel)
         checks.append(check_pt_load)
         checks.append(check_mozglue_order)
